src/Agent.Listener/Agent.cs

```diff
@@ -10,102 +10,121 @@ namespace Microsoft.VisualStudio.Services.Agent.Listener
     [ServiceLocator(Default = typeof(Agent))]
     public interface IAgent : IAgentService
     {
+        CancellationTokenSource TokenSource { get; set; }
         Task<int> ExecuteCommand(CommandLineParser parser);
     }
 
     public sealed class Agent : AgentService, IAgent
     {
+        public CancellationTokenSource TokenSource { get; set; }
```

Comment:
where did you new() the tokensource? i can't find TokenSource=new cancellationTokenSource();